src/curies/discovery.py

```diff
@@ -0,0 +1,93 @@
+"""Discovery new entries for a Converter."""
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, Sequence, Set
+
+from curies import Converter, Record
+
+if TYPE_CHECKING:
+    import rdflib
+
+__all__ = [
+    "discover",
+    "discover_from_rdf",
+]
+
+
+def discover_from_rdf(converter: Converter, graph: "rdflib.Graph", **kwargs: Any) -> Converter:
+    """Discover new URI prefixes from an RDFLib triple store."""
+    uris = set(_yield_uris(graph))
+    return discover(converter, uris, **kwargs)
+
+
+def _yield_uris(graph: "rdflib.Graph") -> Iterable[str]:
+    import rdflib
+
+    for parts in graph.triples((None, None, None)):
+        for part in parts:
+            if isinstance(part, rdflib.URIRef):
+                yield str(part)
+
+
+def discover(
+    converter: Converter,
+    uris: Iterable[str],
+    delimiters: Sequence[str] = "#/",
+    cutoff: int = 30,
+    metaprefix: str = "ns",
+) -> Converter:
+    """Discover new URI prefixes.
+
+    :param converter: A converter with pre-existing definitions. URI prefixes
+        are considered "new" if they can't already be validated by this converter
+    :param uris: An iterable of URIs to search through. Will be taken as a set and
+        each unique entry is only considered once.
+    :param delimiters:
+        The delimiters considered between a putative URI prefix and putative
+        local unique identifier. By default, checks ``#`` first since this is
+        commonly used for URL fragments, then ``/`` since many URIs are constructed
+        with these.
+    :param cutoff: The number of unique URIs with a given prefix needed to call it
+        as unique. This can be adjusted, but is initially high.
```

Comment:
This docstring is very important (`cutoff`) but it is very hard to read - can you make it more clear what the cutoff parameter does?

Reply:
Lots more docs added since